### PR TITLE
Fix rewriter.js and test the API it uses.

### PR DESCRIPTION
**Pull request by cowlicks**
cc @RReverser 

I looked at `rewriter.js`. It was broken :( It happened when the packaged rulesets switched from xml to json in #12273.

I updated `rewriter.js` and did a quick test on a file that just contained `<a href="http://freerangekitten.com/">foo</a>` and it rewrote it as expected.

I also added tests for the API's that rewriter js uses: `addFromJson` (previously `addFromXml`) and `rewriteURI`.

### CODE REVIEW

File: chromium/test/rules_test.js [hunk 1]
```diff
@@ -96,14 +96,35 @@ describe('rules.js', function() {
...
+    });
+
+    describe('#rewriteURI', function() {
+      it('rewrites host added from json', function() {
+        let host = 'freerangekitten.com';
+        this.rsets.addFromJson(rules_json);
+
+        let newuri = this.rsets.rewriteURI('http://' + host + '/', host);
+
+        assert.strictEqual(newuri, 'https://' + host + '/', 'protocol changed to https')
```
Comment (RReverser) [inline chromium/test/rules_test.js]: Could you also add a test for the no-match case, please? (it returns `null` as far as I remember)

File: chromium/test/rules_test.js [hunk 2]
```diff
@@ -96,14 +96,35 @@ describe('rules.js', function() {
   })
 
   describe('RuleSets', function() {
+    let rules_json = JSON.parse('[{"name":"Freerangekitten.com","rule":[{"to":"https:","from":"^http:"}],"target":["freerangekitten.com","www.freerangekitten.com"]}]');
```
Comment (RReverser) [inline chromium/test/rules_test.js]: Perhaps better to inline the object? Result would be the same but would look slightly more readable in the source (and with proper syntax highlighting)

File: chromium/test/rules_test.js [hunk 3]
```diff
@@ -96,14 +96,46 @@ describe('rules.js', function() {
...
+      it('rewrites host added from json', function() {
+        let host = 'freerangekitten.com';
+        this.rsets.addFromJson(rules_json);
+
+        let newuri = this.rsets.rewriteURI('http://' + host + '/', host);
+
+        assert.strictEqual(newuri, 'https://' + host + '/', 'protocol changed to https')
+      })
+
+      it("does'nt rewrite unknown hosts", function() {
```
Comment (RReverser) [inline chromium/test/rules_test.js]: a typo
